js/runner.go

```diff
@@ -318,9 +323,59 @@ func (r *Runner) SetOptions(opts lib.Options) error {
 		r.console = c
 	}
 
+	// FIXME: Resolver probably shouldn't be reset here...
+	// It's done because the js.Runner is created before the full
+	// configuration has been processed, at which point we don't have
+	// access to the DNSConfig, and need to wait for this SetOptions
+	// call that happens after all config has been assembled.
+	// We could make DNSConfig part of RuntimeOptions, but that seems
+	// conceptually wrong since the JS runtime doesn't care about it
+	// (it needs the actual resolver, not the config), and it would
+	// require an additional field on Bundle to pass the config through,
+	// which is arguably worse than this.
+	ttl, err := parseTTL(opts.DNS.TTL.String)
+	if err != nil {
+		return err
+	}
+	dnsSel := opts.DNS.Select.DNSSelect
+	if !dnsSel.IsADNSSelect() {
+		dnsSel = lib.DefaultDNSConfig().Select.DNSSelect
+	}
+	dnsPol := opts.DNS.Policy.DNSPolicy
+	if !dnsPol.IsADNSPolicy() {
+		dnsPol = lib.DefaultDNSConfig().Policy.DNSPolicy
+	}
+	r.Resolver = netext.NewResolver(r.ActualResolver, ttl, dnsSel, dnsPol)
+
```

Comment:
nit: 
I would really prefer if this whole logic was in a separate function somewhere ... 
it looks strange as it is ... I do thin `IsADNSPolicy` should be `.Valid` as in null types ?

Reply:
Sure, I'll give it a try.

Reply:
Resolved by 3f3f454b.